core/trino-main/src/main/java/io/trino/operator/OrderByOperator.java

```diff
@@ -205,6 +205,9 @@ public void finish()
             return;
         }
         checkSuccess(spillInProgress, "spilling failed");
+        if (finishMemoryRevoke.isPresent()) {
+            return;
```

Comment:
Question: could we instead call `finishMemoryRevoke()` here and then proceed without returning? It seems like we're relying on repeated calls to `finish()`, but that it's avoidable.

Reply:
I am keeping this consistent with HBO operator. And chose code to be simpler.

Also note that `finishMemoryRevoke` field became optional, so eventual call to `finishMemoryRevoke` would fail.
The current flow is easier to reason about.